content/en/docs/tasks/debug-application-cluster/inside-container-developing.md

```diff
@@ -0,0 +1,91 @@
+---
+title: Developing and debugging applications running in your cluster
+content_type: task
+---
+
+Kubernetes applications are generally composed of many microservices. Due to environment differences and resource constraints, these microservices somtimes can not run locally as source code. Therefore, developers need to rebuild the image, modify the image version of the workload and then wait for the Pod scheduling after coding. This causes the low efficiency of development in Kubernetes emvironment.
+
+This page describes how you can external tools to debug an application that's running in your cluster.
+
+The tools covered are:
+{{% thirdparty-content %}}
+
+- [Nocalhost](https://nocalhost.dev)
+
+Nocalhost is a tool that enables developers to develop applications in Kubernetes Cluster. It can transfer the remote workload to development mode by using Nocalhost IDE (including VSCode and Jetbrains). When the developer writes code locally, any code changes will be automatically synchronized to the remote development container and initiate the application update immediately (depending on the hot load mechanism of the application or restart the application). The development container inherits all the declarative configurations (configmap, secret, volume, Env) from the original workload, which provides the basis for the application to run in the development container as source code.
```

Comment:
I still believe this post needs some work. For example, the paragraph above could be revised:

```suggestion
Nocalhost is a tool that enables developers to write applications to run on a Kubernetes Cluster.
By using a Nocalhost IDE (for example, VSCode or Jetbrains), you can switch the remote workload
into development mode. When code is updated on your local machine, changes to code are automatically
synchronized to the remote development environment. The code change then automatically triggers
an update to the target application. Depending on the nature of the application, such an update could
be a hot reload or a process restart. The development container where the updated application runs
inherits all the configurations (e.g. ConfigMap, Secret, Volume, Env) from the original container.
```

I'm not sure what a "development container" means and I'm curious how the target workload/application
is automatically restarted.

Reply:
I'm not a native speaker, so the rewriting above only serves an example how the text can be reorganized. It is only an example to show how heavily this post should be revised. Without such a rewriting, this post is difficult to understand.

Reply:
Thank you! 
I have explained about the "development container" and rewritten it according to your comments.